app/controllers/topics_controller.rb

```diff
@@ -16,6 +16,24 @@ def create
     end
   end
 
+  def show
+    @topic = Topic.find_by_id(params[:id])
+
+    if @topic.nil?
+      redirect_to topics_path
+    end
+  end
+
+  def edit
+    @topic = Topic.find_by_id(params[:id])
```

Comment:
Just use the `find` method which defaults to using the id.
```
@topic = Topic.find(params[:id])
```

Reply:
Hmmm... 🤔 
Notice how that line keeps has repeated in four actions inside this controller. Whenever you see something repeating 3 or more times, it's usually a good time to think about DRYing up your code.

In this case, a `before_action` call (previously called a `before_filter` in earlier versions of Rails) will do the job. I'll let you look that up. You could even do it as a separate pull request if you like.